portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles deleting comment data */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  private static final String COMMENT_ENTITY = "Comment";
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private Query commentsQuery = new Query(COMMENT_ENTITY);
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Remove all entities from datastore
+    PreparedQuery results = datastore.prepare(commentsQuery);
+    List<Key> keys = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      keys.add(entity.getKey());
+    }
+    datastore.delete(keys);
```

Comment:
Is there a method to delete keys inline? As in, can you just call `delete(entity.getKey())` inside of the for loop instead of having to maintain an external variable?

Reply:
Looks like there is, thanks!